floop-data-export.py

```diff
@@ -21,34 +26,34 @@
 
 db = firestore.client()
 
+
 # pull conversation documents where 'Comment_Preview' field isn't certain
-#   text, and pulling a limit of 10k.
+#   text, and pulling a limit of 1k.
 conversations = db.collection(
     'Databases/Dev_Database/Conversations').where('Comment_Preview', 'not-in', [
-        'What is your goal for this year?', 'Audio Comment', 'Freeform', 'Freeform Comment', '', ' ']).limit(10000).get()
+        'What is your goal for this year?', 'Audio Comment', 'Freeform', 'Freeform Comment', '', ' ']).limit(n).get()
 
 if __name__ == '__main__':
 
     # Iniialize empty list, "cList"
     cList = []
-# For each item in Conversations, ea entry has a Messages collection
+    
+# For each item in Conversations, each entry has a Messages collection
 # all documents will be ordered by 'Date_Submitted' and only the first comment will be pulled.
-# For each document in the pulled list(currently limited to 1 per collection),
 #    get value of "Text" field and add to cList
 
     for convo in conversations:
         convo_entries = convo.reference.collection(
-            'Messages').order_by(u'Date_Submitted').limit(1).get()
+            'Messages').order_by(u'Date_Submitted').get()
 
         for entry in convo_entries:
-            cList.append(entry.get("Text").strip())
+           # cList.append(entry.get("Text").strip())
+             cList.append({
+                'Text': entry.get("Text").strip()
+                #Sender: "Teacher" | "Student",
```

Comment:
The current format of the code has output that appends `'Text':` in front of each text entry, and they are saved as objects, but doesn't have the uid saved to identify and Teacher/student.  Current output looks like this:
`[{"Text": ""}, {"Text": ""}, {"Text": ""}, {"Text": ""}, {"Text": ""}, {"Text": ""}, {"Text": "plolds asdsdsds sdsds s sdsds sd sd sd"}, {"Text": "f"}, {"Text": ""}, {"Text": ""}, {"Text": "a"}, {"Text": "d"}, {"Text": "f"}, {"Text": "ss"}, {"Text": "xcvcvcvcvxcvcrgrrrgrgrgr"}, {"Text": "d"}, {"Text": "n"}, {"Text": "d"}, {"Text": "g"}, {"Text": "good"},...`

And the code to filter out duplicates for single strings was removed, so updates to filter those out will still be needed.

Reply:
Thanks Mike, I have created a mapping object to classify the sender id into a Teacher or Student.
For duplicate single strings, i have placed a condition check to filter out empty strings.
Please let me know if this isn't what is expected. I have pushed the new code. I shall create a pull request once I get this confirmation from you